### DISCUSSION TITLE
Crazyflie with GPS

### DISCUSSION THREAD

**Discussion by matin7068** [bitcraze/discussions#1459] [General]
I am trying to interface a GPS module with crazyflie 2.1. I tried each and every steps done by previous users to interface, still no output is showing. If someone has done it, can you please explain step by step how to interface it. I am using a simple UBLOX NEO-6M module.

**Comment by ataffanel** [comment]
Can you tell us a bit more what step you are following and what output you expect to see?

GPS support in Crazyflie is very old and experimental. While there is a deck driver that implements NMEA decoding, I do not thing the output data from the decoding are used anywhere in the firmware.

**Reply by matin7068** [reply]
For now I have made some changes in the gtgps.c and config.h file and tried it. But there is no output. Is there any way to just know whether the Drone is getting any signal from the module. First goal is to get any sort of information in the cfclient. I just want to know what steps have others done to interface the gps module.

**Reply by matin7068** [reply]
Additionally, while flashing the firmware  (by customising the firmware folder and compressing to zip file) through cfclient in bootloader mode, whenever i click on "program", there is no progress bar status (the progress bar should be completed twice as far as i know). This issue doesn't occur when i update the firmware from release/ available downloads. This is another problem I am facing these days.  The drone just restarts, and when I check the version, It doesnt get changed to the one I flashed.

Is there some other steps to be performed while flashing "from files"?

**Reply by evoggy** [reply]
Hi! You can flash the STM32 firmware [directly from the command line](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/#using-crazyradio). The easiest way to get a feel for if it's working might be to do some debug printing, have a look at how the [DEBUG_PRINT macro is being used](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/howto/#writing-the-driver).
